reconcile/utils/oc.py

```diff
@@ -267,7 +267,8 @@ def get_api_resources(self):
         return [r.split()[-1] for r in results]
 
     def get_version(self):
-        cmd = ['version']
+        # this is actually a 10 second timeout, because: oc reasons
+        cmd = ['version', '--request-timeout=5']
```

Comment:
out of curiosity, how long it takes otherwise?
@arilivigni ptal

Reply:
this was taking 10-15 minutes per run locally, as several of the clusters aren't or weren't reachable (when tunnel is off). this also happens in our production code anywhere oc is used and a cluster is not reachable by OC. in nearly all cases a response comes back normally under 4 seconds. the timeout options in the oc client are not granular. also (an openshift "feature"): the server can respond back with a retry timeout that the client honors.. so any cluster can say: hey, there was an error, but retry in 60 seconds. it's a bit of a mess, but a ~10 second timeout (--request-timeout=5) fit the 98 percentile of connect cases observed. this version check also is invoked even if the don't-init-resources parameter is passed: https://github.com/app-sre/qontract-reconcile/blob/42d8ec48b20f5fef7b29bd24ec20397bff44efa4/reconcile/utils/oc.py#L77-L78  

development observation: this didn't have any debug-visbility around it, as the oc call is a silent exec().. so for a while i thought it was either vault or my original usage of a prom lib causing long run times. i would very much like to replace all usage of the oc binary with the maintained openshift py client (which wraps the kube client): https://github.com/openshift/openshift-client-python. obviously the scope of that is well outside this effort. :)

Reply:
@arilivigni  this make sense to you? ok to resolve?

Reply:
:+1:  sounds good to me